joomla/joomla-cms
# [4.4] Fix newsfeed category html view

### Summary of Changes

Fix newsfeed category html view as down-merge from 5.1-dev. This is a sub-part from https://github.com/joomla/joomla-cms/pull/43722 to use joomla-cypress 1.1.1 in 4.4-dev too.

### Testing Instructions

The warning is displayed in A) the frontenend view and B) the System Test (with a speaking `checkForPhpNoticesOrWarnings`). Verify the warning before patching. After patching, check that there are no more warnings.

### Actual result BEFORE applying this Pull Request

#### A) Manual Test
Requesting URL `/index.php?option=com_newsfeeds&view=category&id=5` in fresh 4.4-dev shows the PHP Warning message (id comes from Components | News Feeds | Categories | Uncategorised):
<img width="744" alt="shoot" src="https://github.com/user-attachments/assets/7949e702-f5db-4a0b-96ae-bfe4f7e19969">

#### B) System Test
As preparation install speaking `checkForPhpNoticesOrWarnings` in overwriting lines 74 ... 89 in `node_modules/joomla-cypress/src/support.js`with the following lines:
```
  Cypress.Commands.add('checkForPhpNoticesOrWarnings', () => {
    cy.log('**Check for PHP notices and warnings**')

    cy.document().then((doc) => {
      const pageSource = doc.documentElement.innerHTML
      // Search for PHP problem keywords in bold style with colon and collect the found keyword and the message
      const regex = /<b>(Warning|Deprecated|Notice|Strict standards)<\/b>:(.*?)(<br|$)/
      const match = regex.exec(pageSource)
      if (match) {
        // Directly fail with the reason, the keyword found and report the PHP problem message, e.g.
        // Error: Unwanted PHP Warning: "Attempt to read property \"id\" on null in <b>/joomla-cms/components/com_newsfeeds/src/View/Category/HtmlView.php</b> on line <b>92</b>"
        throw new Error(`Unwanted PHP ${match?.[1]}: ${JSON.stringify(match?.[2])}`)
      }
    })

    cy.log('--Check for PHP notices and warnings--')
  })
  ```

Run the test specifications:
```
npx cypress run --spec tests/System/integration/site/components/com_newsfeed/Category.cy.js
```
Fails with:
```
  Running:  Category.cy.js                                                                  (1 of 1)

  Test in frontend that the newsfeeds category view
    ✓ can display a list of feeds in a menu item (539ms)
    1) "after each" hook for "can display a list of feeds without a menu item"

  1 passing (1s)
  1 failing

  1) Test in frontend that the newsfeeds category view
       "after each" hook for "can display a list of feeds without a menu item":
     Error: Unwanted PHP Warning: "  Attempt to read property \"id\" on null in <b>/Users/hlu/Desktop/no_backup/joomla-cms/components/com_newsfeeds/src/View/Category/HtmlView.php</b> on line <b>92</b>"
```

💡 With overwritten speaking `checkForPhpNoticesOrWarnings`, if you run the entire system tests, there are four failures before this PR and only three after. The remaining three errors are fixed by https://github.com/joomla/joomla-cms/pull/43880 and https://github.com/joomla/joomla-cms/pull/43881.

### Expected result AFTER applying this Pull Request

No warnings in A) frontend view or B) System Test.

### Link to documentations
Please select:


### Up-Merging

@LadySolveig @bembelimen there is no up-merge required as it is only a down-merge from 5.1-dev
